doc: Add steps for Transifex to release process (#21440)
bitcoin/bitcoin: doc/release-process.md

  - This update should be reviewed with a reindex-chainstate with assumevalid=0 to catch any defect
     that causes rejection of blocks in the past history.
- Clear the release notes and move them to the wiki (see "Write the release notes" below).
- Translations on Transifex
    - Create [a new resource](https://www.transifex.com/bitcoin/bitcoin/content/) named after the major version with slug `[bitcoin.qt-translation-<RRR>x]` where `RRR` is the major branch number padded with zeros. Use `src/qt/locale/bitcoin_en.ts` to create it.
    - Copy over the translations of the previous version using [transifex-migrate-resource](https://github.com/bitcoin-core/bitcoin-maintainer-tools#transifex-migrate-resource).

>     - Update the transifex slug in [`.tx/config`](/.tx/config) to the slug of the resource created in first step. This identifies which resource translations will be synchronized from.

Reviewer: nit article usage
```suggestion
    - Update the transifex slug in [`.tx/config`](/.tx/config) to the slug of the resource created in the first step. This identifies which resource translations will be synchronized from.
```

>     - Update the transifex slug in [`.tx/config`](/.tx/config) to the slug of the resource created in first step. This identifies which resource translations will be synchronized from.

Reviewer 2: nit:
```suggestion
    - Update the Transifex slug in [`.tx/config`](/.tx/config) to the slug of the resource created in first step. This identifies which resource translations will be synchronized from.
```

    - Make an announcement that translators can start translating for the new version. You can use one of the [previous announcements](https://www.transifex.com/bitcoin/bitcoin/announcements/) as a template.

>     - Change the resource auto-update URL for the resource to `master`, e.g. `https://raw.githubusercontent.com/bitcoin/bitcoin/<branch>/src/qt/locale/bitcoin_en.ts`. (Do this only after the previous steps to prevent an auto-update from interfering.

Reviewer: - nit on `resource`, seems to be used redundantly
- nit, e.g normally is followed by a comma
- missing closing parentheses
```suggestion
    - Change the auto-update URL for the resource to `master`, e.g., `https://raw.githubusercontent.com/bitcoin/bitcoin/<branch>/src/qt/locale/bitcoin_en.ts`. (Do this only after the previous steps to prevent an auto-update from interfering.)
```

>     - Change the resource auto-update URL for the resource to `master`, e.g. `https://raw.githubusercontent.com/bitcoin/bitcoin/<branch>/src/qt/locale/bitcoin_en.ts`. (Do this only after the previous steps to prevent an auto-update from interfering.

Reviewer 2: ```suggestion
    - Change the resource auto-update URL for the resource to `master`, i.e., `https://raw.githubusercontent.com/bitcoin/bitcoin/master/src/qt/locale/bitcoin_en.ts`. (Do this only after the previous steps to prevent an auto-update from interfering.
```
